OutOfSchool/OutOfSchool.WebApi/Services/ParentService.cs

```diff
@@ -0,0 +1,108 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using OutOfSchool.Services.Models;
+using OutOfSchool.Services.Repository;
+using OutOfSchool.WebApi.Mapping.Extensions;
+using OutOfSchool.WebApi.Models;
+
+namespace OutOfSchool.WebApi.Services
+{
+    /// <summary>
+    /// Service with business logic for ParentController.
+    /// </summary>
+    public class ParentService : IParentService
+    {
+        private readonly IEntityRepository<Parent> repository;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ParentService"/> class.
+        /// </summary>
+        /// <param name="entityRepository">Repository for some entity.</param>
+        public ParentService(IEntityRepository<Parent> entityRepository)
+        {
+            this.repository = entityRepository;
+        }
+
+        /// <inheritdoc/>
+        public async Task<ParentDTO> Create(ParentDTO parent)
+        {
+            CreateValidation(parent);
+            Parent res = await repository.Create(parent.ToDomain()).ConfigureAwait(false);
+            return res.ToModel();
+        }
+
+        /// <inheritdoc/>
+        public async Task Delete(long id)
+        {
+            Parent parent = await repository.GetById(id).ConfigureAwait(false);
+            if (parent == null)
+            {
+                throw new ArgumentException("This id doesn't exist", nameof(id));
+            }
+
+            await repository.Delete(parent).ConfigureAwait(false);
+        }
+
+        /// <inheritdoc/>
+        public async Task<IEnumerable<ParentDTO>> GetAll()
+        {
+            IEnumerable<Parent> parents = await this.repository.GetAll().ConfigureAwait(false);
+            return parents.Select(x => x.ToModel()).ToList();
+        }
+
+        /// <inheritdoc/>
+        public async Task<ParentDTO> GetById(long id)
+        {
+            Parent parent = await repository.GetById((int)id).ConfigureAwait(false);
+            if (parent == null)
+            {
+                throw new ArgumentException("Not Found", nameof(id));
+            }
+
+            return parent.ToModel();
+        }
+
+        /// <inheritdoc/>
+        public async Task<ParentDTO> Update(ParentDTO parent)
+        {
+            await UpdateValidation(parent).ConfigureAwait(false);
+            Parent res = await repository.Update(parent.ToDomain()).ConfigureAwait(false);
+            return res.ToModel();
+        }
+
+        public static void CreateValidation(ParentDTO parent)
```

Comment:
do we need this validation? Regex are responsible for these validations. Even data isn't valid, it will not get to this method.